tests/integration/helm/util.go

```diff
@@ -55,34 +56,108 @@ const (
 	IngressReleaseName     = "istio-ingress"
 	ControlChartsDir       = "istio-control"
 	GatewayChartsDir       = "gateway"
+	CniChartsDir           = "istio-cni"
+	ZtunnelChartsDir       = "ztunnel"
+	RepoCniChartPath       = "cni"
+	CniReleaseName         = ReleasePrefix + "cni"
+	RepoZtunnelChartPath   = "ztunnel"
+	ZtunnelReleaseName     = "ztunnel"
 
 	RetryDelay   = 2 * time.Second
 	RetryTimeOut = 5 * time.Minute
 	Timeout      = 2 * time.Minute
+
+	defaultValues = `
+global:
+  hub: %s
+  tag: %s
+revision: "%s"
+`
+	// TODO: Remove this once the previous release version for the ambient upgrade test becomes 1.21, and start using --set profile=ambient
+	ambientProfileOverride = `
+global:
+  hub: %s
+  tag: %s
+variant: ""
+meshConfig:
+  defaultConfig:
+    proxyMetadata:
+      ISTIO_META_ENABLE_HBONE: "true"
+pilot:
+  variant: ""
+  env:
+    # Setup more secure default that is off in 'default' only for backwards compatibility
+    VERIFY_CERTIFICATE_AT_CLIENT: "true"
+    ENABLE_AUTO_SNI: "true"
+
+    PILOT_ENABLE_HBONE: "true"
+    CA_TRUSTED_NODE_ACCOUNTS: "istio-system/ztunnel,kube-system/ztunnel"
+    PILOT_ENABLE_AMBIENT_CONTROLLERS: "true"
+cni:
+  logLevel: info
+  privileged: true
+  ambient:
+    enabled: true
+
+  # Default excludes istio-system; its actually fine to redirect there since we opt-out istiod, ztunnel, and istio-cni
+  excludeNamespaces:
+    - kube-system
+`
 )
 
 // ManifestsChartPath is path of local Helm charts used for testing.
 var ManifestsChartPath = filepath.Join(env.IstioSrc, "manifests/charts")
 
+// getValuesOverrides returns the values file created to pass into Helm override default values
+// for the hub and tag
+func GetValuesOverrides(ctx framework.TestContext, hub, tag, revision string, isAmbient bool) string {
```

Comment:
If we put this in testContext, I assume we currently set it for ambient profile, and then this test alone run under the ambient integration suite, instead of helm suite, as it is currently done.. I felt it is better the way it is now to consolidate all helm tests together?

Reply:
Ah, ok. We might eventually have to rethink that "ambient suite" versus "everything else suite" divide or we are gonna have a whole lot of `testHelper(....., isAmbient bool)` signatures all over the test suite, but I admit that's out of scope for this.